src/main/java/maksym/fedorenko/bookstore/model/Book.java

```diff
@@ -57,4 +59,7 @@ public class Book {
             inverseJoinColumns = @JoinColumn(name = "category_id")
     )
     private Set<Category> categories = new HashSet<>();
+
+    @OneToMany(mappedBy = "book", cascade = CascadeType.REMOVE)
```

Comment:
I assume you also need to specify `orphanRemoval`, since cartItem can't exist without a book, right?